plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSnowable.java

```diff
@@ -2,52 +2,46 @@
 
 import com.denizenscript.denizen.objects.MaterialTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.ElementTag;
-import com.denizenscript.denizencore.objects.properties.Property;
-import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import org.bukkit.block.data.BlockData;
 import org.bukkit.block.data.Snowable;
 
-public class MaterialSnowable implements Property {
+public class MaterialSnowable extends MaterialProperty<ElementTag> {
 
-    public static boolean describes(ObjectTag material) {
-        return material instanceof MaterialTag
-                && ((MaterialTag) material).hasModernData()
-                && ((MaterialTag) material).getModernData() instanceof Snowable;
-    }
+    // <--[tag]
+    // @object MaterialTag
+    // @name snowy
+    // @input ElementTag(Boolean)
+    // @description
+    // Controls whether this material is covered in snow.
+    // -->
 
-    public static MaterialSnowable getFrom(ObjectTag _material) {
-        if (!describes(_material)) {
-            return null;
-        }
-        else {
-            return new MaterialSnowable((MaterialTag) _material);
-        }
+    public static boolean describes(MaterialTag material) {
+        BlockData data = material.getModernData();
+        return data instanceof Snowable;
     }
 
-    public static final String[] handledMechs = new String[] {
-            "snowy"
-    };
+    MaterialTag material;

```

Comment:
why is this here?

Reply:
This code would produce an error, this value is read from but never set, indicating you didn't actually test this PR.